File: solutions/ts/03.ts
function drawGift(size: number, symbol: string): string {
  if (size < 2) return "";
  let box = "";
  box += symbol.repeat(size) + "\n";
  for (let i = 1; i < size - 1; i++) {
    box += symbol + " ".repeat(size - 2) + symbol + "\n";
  }
  box += symbol.repeat(size);
  return box;
}

const g1 = drawGift(4, "*");
console.log(g1);
/*
 ****
 *  *
 *  *
 ****
 */

const g2 = drawGift(3, "#");
console.log(g2);
/*
###
# #
###
*/

const g3 = drawGift(2, "-");
console.log(g3);
/*
--
--
*/

const g4 = drawGift(1, "+");
console.log(g4);
// ""  poor intern…

/* 
  Score: ★★★★★★★★ (3 bonus stars for solving it today!)
  Points: 360
  ops: 4275
*/

File: solutions/python/03.py
def draw_gift(size, symbol):
    if size < 2:
        return ''
    box = ""
    box += symbol * size + "\n"
    for _ in range(1, size-1):
        box += symbol + " " * (size-2) + symbol + "\n"
    box += symbol * size
    return box


g1 = draw_gift(4, "*")
print(g1)
"""
 ****
 *  *
 *  *
 ****
"""

g2 = draw_gift(3, "#")
print(g2)
"""
###
# #
###
"""

g3 = draw_gift(2, "-")
print(g3)
"""
--
--
"""

g4 = draw_gift(1, "+")
print(g4)
# ""  poor intern…

"""
  Score: ★★★★★★★ (3 bonus stars for solving it today!)
  Points: 460
  ops: 4713
"""
